src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockWorkspaceServer.scala

```diff
@@ -276,7 +277,7 @@ object MockWorkspaceServer {
         request()
           .withMethod("GET")
           .withPath(s"${workspaceBasePath}/%s/%s/submissions/%s/workflows/%s"
-            .format(mockSpacedWorkspace.namespace, mockSpacedWorkspace.name, mockValidId, mockValidId)))
+            .format(mockSpacedWorkspace.namespace, UrlEscapers.urlPathSegmentEscaper().escape(mockSpacedWorkspace.name), mockValidId, mockValidId)))
```

Comment:
I see you only had to do this URL escaping once in this PR...but it also looks pretty ugly -- if you had to do this 20 times, once for each test case, what would your solution look like?

I'm asking not because I want you to implement that solution (since this is just a one-off) but just curious what it'd look like in Scala :)

Reply:
I would take one of two routes:

wrap the escaping in something terser, like [`StreamingPassthrough.escapePathSegment`](https://github.com/broadinstitute/firecloud-orchestration/blob/91ad97a82008ca8742e5a63b23d2bc1984f10188/src/main/scala/org/broadinstitute/dsde/firecloud/utils/StreamingPassthrough.scala#L32-L34), so you end up with:
```scala
          .withPath(s"${workspaceBasePath}/%s/%s/submissions/%s/workflows/%s"
            .format(mockSpacedWorkspace.namespace, escapePathSegment(mockSpacedWorkspace.name), mockValidId, mockValidId)))
```
this isn't a big win, but it's nicer.

Alternately, I would make an implicit class that provides methods for strings; then, I could call my custom method directly on the thing I wanted to escape:
```scala
    implicit class StringEscaper(s: String) {
      def escape: String = UrlEscapers.urlPathSegmentEscaper().escape(s)
    }

…
          .withPath(s"${workspaceBasePath}/%s/%s/submissions/%s/workflows/%s"
            .format(mockSpacedWorkspace.namespace, mockSpacedWorkspace.name.escape, mockValidId, mockValidId)))


```